src/boost_histogram/_internal/hist.py

```diff
@@ -851,37 +851,72 @@ def __getitem__(self: H, index: IndexingExpr) -> H | float | Accumulator:
 
+            groups = []
             if ind != slice(None):
                 merge = 1
                 if ind.step is not None:
-                    if hasattr(ind.step, "factor"):
+                    if getattr(ind.step, "factor", None) is not None:
                         merge = ind.step.factor
                     elif callable(ind.step):
                         if ind.step is sum:
                             integrations.add(i)
+                        elif getattr(ind.step, "groups", None) is not None:
+                            groups = ind.step.groups
                         else:
-                            raise RuntimeError("Full UHI not supported yet")
+                            raise NotImplementedError
 
                         if ind.start is not None or ind.stop is not None:
                             slices.append(
                                 _core.algorithm.slice(
                                     i, start, stop, _core.algorithm.slice_mode.crop
                                 )
                             )
-                        continue
+                        if len(groups) == 0:
+                            continue
                     else:
                         raise IndexError(
                             "The third argument to a slice must be rebin or projection"
                         )
 
                 assert isinstance(start, int)
                 assert isinstance(stop, int)
-                slices.append(_core.algorithm.slice_and_rebin(i, start, stop, merge))
+                # rebinning with factor
+                if len(groups) == 0:
+                    slices.append(
+                        _core.algorithm.slice_and_rebin(i, start, stop, merge)
+                    )
+                # rebinning with groups
+                elif len(groups) != 0:
+                    reduced = self._hist
+                    axes = [reduced.axis(x) for x in range(reduced.rank())]
+                    reduced_view = reduced.view(flow=True)
+                    new_axes_indices = [axes[i].edges[0]]
+
+                    j: int = 0
+                    for group in groups:
+                        new_axes_indices += [axes[i].edges[j + 1 : j + group + 1][-1]]
+                        j += group
+
+                    variable_axis = Variable(
+                        new_axes_indices, metadata=axes[i].metadata
+                    )
+                    axes[i] = variable_axis._ax
+                    reduced_view = np.take(
+                        reduced_view, range(len(reduced_view)), axis=i
+                    )
+
+                    logger.debug("Axes: %s", axes)
+
+                    new_reduced = reduced.__class__(axes)
+                    new_reduced.fill(
+                        np.add.reduceat(reduced.to_numpy()[0], new_axes_indices, axis=i)
+                    )
+                    reduced = new_reduced
```

Comment:
I am doing something wrong here; I can't figure out how to distribute the bin contents once bins are regrouped. Could someone please help me with this? Thanks!

Current status:
```py
In [1]: import boost_histogram as bh
   ...: import numpy as np
   ...: h = bh.Histogram(bh.axis.Regular(10, 0, 1))
   ...: h.fill(np.random.normal(size=1_000_000))
   ...: rebin = bh.tag.Rebinner(factor=2)
   ...: h[::rebin]  # rebin with a factor
Out[1]: Histogram(Regular(5, 0, 1), storage=Double()) # Sum: 342342.0 (1000000.0 with flow)

In [2]: rebin = bh.tag.Rebinner(groups=[1, 2, 3])

In [3]: h[::rebin] # rebin into groups (variable axis)
Out[3]: Histogram(Variable([0, 0.1, 0.3, 0.6], metadata=...), storage=Double()) # Sum: 0.0 (4.0 with flow)

In [4]: print(h[::rebin])  # doing something wrong while distributing the values
              ┌──────────────────────────────────────────────────────────────┐
[-inf,   0) 0 │                                                              │
[   0, 0.1) 0 │                                                              │
[ 0.1, 0.3) 0 │                                                              │
[ 0.3, 0.6) 0 │                                                              │
[ 0.6, inf) 4 │█████████████████████████████████████████████████████████████ │
              └──────────────────────────────────────────────────────────────┘

In [5]: rebin = bh.tag.Rebinner(groups=[1 for x in range(10)]) # rebin into same bins but introduce a variable axis

In [6]: h[::rebin] # rebin into groups (variable axis)
Out[6]: Histogram(Variable([0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1], metadata=...), storage=Double()) # Sum: 0.0 (11.0 with flow)

In [7]: print(h[::rebin])  # doing something wrong while distributing the values
               ┌─────────────────────────────────────────────────────────────┐
[-inf,   0) 0  │                                                             │
[   0, 0.1) 0  │                                                             │
[ 0.1, 0.2) 0  │                                                             │
[ 0.2, 0.3) 0  │                                                             │
[ 0.3, 0.4) 0  │                                                             │
[ 0.4, 0.5) 0  │                                                             │
[ 0.5, 0.6) 0  │                                                             │
[ 0.6, 0.7) 0  │                                                             │
[ 0.7, 0.8) 0  │                                                             │
[ 0.8, 0.9) 0  │                                                             │
[ 0.9,   1) 0  │                                                             │
[   1, inf) 11 │████████████████████████████████████████████████████████████ │
               └─────────────────────────────────────────────────────────────┘
```

cc: @matthewfeickert @andrzejnovak @alexander-held

Reply:
Just birdseye and I only usually interact with `bh` with `hist`, but on L911 you're calling `fill`? I'd expect there be some value assignment rather than filling. As the yields in your demo hist are >1, would be consistent with them appearing in overflow on an 0,1 axis.

Reply:
Yes, maybe I am missing an internal function that can be used here?